Tools/Scripts/webkitpy/common/checkout/checkout.py

```diff
@@ -47,6 +48,27 @@
 # FIXME: Move a bunch of ChangeLog-specific processing from SCM to this object.
 # NOTE: All paths returned from this class should be absolute.
 class Checkout(object):
+    COMMIT_SUBJECT_RE = re.compile(b'Subject: \[PATCH ?(\d+\/\d+)?] (.+)')
+    FILTER_BRANCH_PROGRAM = '''import re
+import sys
+
+lines = [l for l in sys.stdin]
+for s in re.split(r' (Need the bug URL \(OOPS!\).)|(\S+:\/\/\S+)', lines[0].rstrip()):
+    if s and s != ' ':
+        print(s)
+for l in lines[1:]:
+    sys.stdout.write(l)
+'''
+
+    @classmethod
+    def filter_patch_content(cls, content, reviewer=None):
```

Comment:
Do we guarantee content is always passed in as bytes? Otherwise, we will have exception on line 69 when concat bytes with string.

Reply:
Good point, will explicitly encode content.